src/storage/src/hummock/sstable_store.rs

```diff
@@ -497,6 +447,63 @@ impl SstableStore {
 
 pub type SstableStoreRef = Arc<SstableStore>;
 
+impl MemoryCollector for SstableStore {
+    fn get_meta_memory_usage(&self) -> u64 {
+        self.meta_cache.get_memory_usage() as u64
+    }
+
+    fn get_data_memory_usage(&self) -> u64 {
+        self.block_cache.size() as u64
+    }
+
+    // TODO: limit shared-buffer uploading memory
+    fn get_total_memory_usage(&self) -> u64 {
+        0
+    }
+}
+
+#[async_trait::async_trait]
+pub trait SstableStoreWrite: Send + Sync {
```

Comment:
What's the purpose of this trait? Only one implementation,  the `SstableStore`, for it.